### PR TITLE
WIP: feat: add userdata api to SDK

### PR DESCRIPTION
**Pull request by abarbare**
Hello!
This PR implement the reading of `user_data` information from the SDK. As this feature need to call the `base_url` with a privileged port, I had to override the python requests object. 

Regarding the unit tests, I didn't find an elegant way to integrate them except mocking get `get_userdata` method and compare the result to a string (which is basically comparing two static strings). If you have any piece of advice about it, I would be more than happy to integrate them.

### CODE REVIEW

File: scaleway/apis/api_userdata.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+        If `as_shell` is True, return a string easily parsable by a shell. If
+        False, return a dictionary.
+        """
+        with requests.Session() as s:
+            s.mount('http://', SourcePortAdapter(random.randrange(1, 1024)))
+
+            if as_shell:
+                return s.get('%s/user_data' % self.base_url).text
+            return s.get('%s/user_data?format=json' % self.base_url).text
```
Comment (bchatelard) [inline scaleway/apis/api_userdata.py]: You might want to return a dict here, not `.text`

File: scaleway/apis/api_userdata.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    base_url = 'http://169.254.42.42/'
+
+    def __init__(self, **kwargs):
+
+        assert 'auth_token' not in kwargs, \
+            'auth_token is not required to query the metadata API'
+
+        super(UserDataAPI, self).__init__(auth_token=None, **kwargs)
+
+    def get_userdata(self, as_shell=False):
```
Comment (bchatelard) [inline scaleway/apis/api_userdata.py]: Actually this calls returns the list of userdata keys, not the values.

File: scaleway/apis/api_userdata.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        """ Returns server userdata.
+
+        If `as_shell` is True, return a string easily parsable by a shell. If
+        False, return a dictionary.
+        """
+        with requests.Session() as s:
+            s.mount('http://', SourcePortAdapter(random.randrange(1, 1024)))
```
[the remark this answers is not in the document]
Comment (abarbare) [reply scaleway/apis/api_userdata.py]: Is there any limit on the number of variables defined by a user?
As the `/user_data` return only the keys, we have to loop over each one to get the value so we can get a most 1023 values.

**Comment by abarbare** [comment]
How do you want to handle the multiline values for a key with the `as_shell` option?
`/conf` generate somehow a line with a header followed by the values 

```
LOCATION='PLATFORM_ID HYPERVISOR_ID NODE_ID CLUSTER_ID ZONE_ID'
LOCATION_PLATFORM_ID=XX
LOCATION_HYPERVISOR_ID=XX
LOCATION_NODE_ID=XX
```
LOCATION_CLUSTER_ID=XX
LOCATION_ZONE_ID=par1
```

**Pull request closed without merging**